google/cloud/storage/internal/async/reader_connection_factory_test.cc

```diff
@@ -0,0 +1,247 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "google/cloud/storage/internal/async/reader_connection_factory.h"
+#include "google/cloud/storage/internal/grpc/object_request_parser.h"
+#include "google/cloud/storage/internal/object_requests.h"
+#include "google/cloud/testing_util/is_proto_equal.h"
+#include <google/protobuf/text_format.h>
+#include <gmock/gmock.h>
+
+namespace google {
+namespace cloud {
+namespace storage_internal {
+GOOGLE_CLOUD_CPP_INLINE_NAMESPACE_BEGIN
+namespace {
+
+using ::google::cloud::testing_util::IsProtoEqual;
+using ::google::protobuf::TextFormat;
+
+TEST(AsyncReaderConnectionFactory, UpdateGenerationDefault) {
+  google::storage::v2::ReadObjectRequest expected;
+  EXPECT_TRUE(TextFormat::ParseFromString(
+      R"pb(
+        bucket: "projects/_/buckets/test-bucket" object: "test-object"
+      )pb",
+      &expected));
+  auto request =
+      storage::internal::ReadObjectRangeRequest("test-bucket", "test-object");
+  auto actual = ToProto(request).value();
+  EXPECT_THAT(actual, IsProtoEqual(expected));
+  UpdateGeneration(actual, storage::Generation());
+  EXPECT_TRUE(TextFormat::ParseFromString(
+      R"pb(
+        bucket: "projects/_/buckets/test-bucket" object: "test-object"
+      )pb",
+      &expected));
```

Comment:
It seems less clear to re-parse when we want the same result (and below).

Reply:
Removed